app/javascript/components/ProductEdit/ProductTab/DefaultDiscountCodeSelector.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from "react";
+
+import { searchProductOfferCodes } from "$app/data/offer_code";
+import { assertResponseError } from "$app/utils/request";
+
+import { ComboBox } from "$app/components/ComboBox";
+import { ProductEditContext, AvailableDiscountCode } from "$app/components/ProductEdit/state";
+import { showAlert } from "$app/components/server-components/Alert";
+import { ToggleSettingRow } from "$app/components/SettingRow";
+import { useDebouncedCallback } from "$app/components/useDebouncedCallback";
+
+export const DefaultDiscountCodeSelector = () => {
+  const context = React.useContext(ProductEditContext);
+  if (!context?.availableDiscountCodes) return null;
+
+  const { uniquePermalink, product, updateProduct, availableDiscountCodes, setAvailableDiscountCodes } = context;
+
+  const defaultOfferCodeId = product.default_offer_code_id;
+  const isEnabled = !!defaultOfferCodeId;
+
+  const selectedDiscountCode = availableDiscountCodes.find((code) => code.id === defaultOfferCodeId) ?? null;
+
+  const formatLabel = (code: AvailableDiscountCode) => code.name || code.code;
+
+  const [query, setQuery] = React.useState<string>(() =>
+    selectedDiscountCode ? formatLabel(selectedDiscountCode) : "",
+  );
+  const [options, setOptions] = React.useState<AvailableDiscountCode[]>(availableDiscountCodes);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isTogglePending, setIsTogglePending] = React.useState(false);
```

Comment:
Without isTogglePending, the dropdown would instantly collapse because there’s no default_offer_code_id yet. This bit of state keeps the form open during that blank period so the seller can type and select the code before we persist it.